Lesson_27_Arrays/js/homework.js

```diff
@@ -0,0 +1,177 @@
+  {
+    name: "Banana",
+    bought: false,
+    amount: 4,
+  },
+  {
+    name: "Avokado",
+    bought: true,
+    amount: 2,
+  },
+  {
+    name: "Oil",
+    bought: true,
+    amount: 1,
+  },
+  {
+    name: "Water",
+    bought: false,
+    amount: 1,
+  },
+  {
+    name: "Shampoo",
+    bought: false,
+    amount: 4,
+  },
+];
+
+//Вывод всего списка на экран таким образом, чтобы сначала шли некупленные продукты, а потом – купленные.
+function sortByStatus(arr) {
+  arr.sort((a, b) => {
+    if (a.bought === b.bought) {
+      return 0;
+    }
+    return b.bought ? -1 : 1;
+  });
+}
+
+sortByStatus(products);
+
+console.log(products);
+
+// Добавление покупки в список. Учтите, что при добавлении покупки с уже существующим в списке продуктом,
+// необходимо увеличивать количество в существующей покупке, а не добавлять новую.
+
+// Не понимаю что хотят ):
+
+function addProductToList(arr, productName) {
+  const existingProduct = arr.find((product) => product.name === productName);
+
+  if (existingProduct) {
+    existingProduct.amount++;
+  } else {
+    arr.push({
+      name: productName,
+      bought: false,
+      amount: 1,
+    });
+  }
+}
+
+addProductToList(products, "Shampoo");
+console.log(products);
+addProductToList(products, "Potatoes");
+console.log(products);
+
+// Покупка продукта. Функция принимает название продукта и отмечает его как купленный.
+function boughtProduct(arr, productName) {
+  const existingProduct = arr.find((product) => product.name === productName);
+
+  if (existingProduct) {
+    existingProduct.bought = true;
+  }
+}
+boughtProduct(products, "Shampoo");
+console.log(products);
+//2. Создать массив, описывающий чек в магазине. Каждый элемент массива состоит из названия товара,
+// количества и цены за единицу товара. Написать следующие функции:
+const receipt = [
+  { name: "banana", number: 5, price: 4 }, //product etot wes bolszoj (eto dla sebia)
+  { name: "shampoo", number: 2, price: 10 },
+  { name: "kiwi", number: 4, price: 8 },
+  { name: "knife", number: 1, price: 40 },
+];
+
+// Распечатка чека на экран;
+function printReceipt(arr) {
+  let result = "";
+  for (const product of arr) {
+    result += `Product: ${product.name} × ${product.number} - $${product.price}\n`;
+  }
+  return result;
+}
+
+console.log(printReceipt(receipt));
+
+// Подсчет общей суммы покупки;
+function totalAmount(arr) {
+  let sum = 0;
+  for (const product of arr) {
+    sum += product.price * product.number;
+  }
+  return sum;
+}
+console.log(totalAmount(receipt));
+
+// Получение самой дорогой покупки в чеке;
+function expensiveProduct(arr) {
+  let max = 0;
+
+  for (const product of arr) {
+    const cost = product.price * product.number;
+
+    if (cost > max) {
+      max = cost;
+    }
+  }
+  return max;
+}
+console.log(expensiveProduct(receipt));
+
+// Подсчет средней стоимости одного товара в чеке.
+
+function averageCost(arr) {
+  let sum = 0;
+  for (const product of arr) {
+    sum += product.price;
```

Comment:
тут нужно считать стоимость
```suggestion
    sum += product.price * product.number;
```